packages/simple-xcm/simplexcm.ts

```diff
@@ -253,25 +257,39 @@ export class SimpleXcm {
     }
   }
 
-  async estimateExtrinsicXcmFees(
+  async tryEstimateExtrinsicXcmFees(
     origin: Origin,
     xt: SubmittableExtrinsic<'promise'>,
     feeAssetId: AssetId,
-  ) {
-    const estimatedFees = await this.estimator.estimateExtrinsicFees(
-      origin,
-      xt,
-      feeAssetId,
-      {
-        estimatorResolver: (universalLocation: InteriorLocation) => {
-          const chainInfo =
-            this.registry.chainInfoByUniversalLocation(universalLocation);
-          return Estimator.connect(chainInfo);
+  ): Promise<{value: bigint} | {error: TooExpensiveFeeError}> {
+    try {
+      const estimatedFees = await this.estimator.estimateExtrinsicFees(
+        origin,
+        xt,
+        feeAssetId,
+        {
+          estimatorResolver: (universalLocation: InteriorLocation) =>
+            Estimator.connect(
+              this.registry.chainInfoByUniversalLocation(universalLocation),
+            ),
         },
-      },
-    );
-
-    return estimatedFees;
+      );
+      return {value: estimatedFees};
+    } catch (errors) {
+      if (errors instanceof FeeEstimationErrors) {
+        const totalValue = errors.errors.reduce((sum, error) => {
+          if (error.cause instanceof TooExpensiveFeeError) {
+            return sum + BigInt(error.cause.missingAmount);
```

Comment:
The `missingAmount` is already a bigint according to the `TooExpensiveFeeError` declaration

```suggestion
            return sum + error.cause.missingAmount;
```